src/segment-loader.js

```diff
@@ -1515,6 +1515,10 @@ export default class SegmentLoader extends videojs.EventTarget {
       // the segment has a map
       this.appendInitSegment_[type] = map ? true : false;
 
+      // we need to clear out the fmp4 active init segment id, since
+      // we are appending the muxer init segment
+      this.activeInitSegmentId_ = null;
```

Comment:
This section of this function only really runs if we transmuxed from a ts to an fmp4. The problem: Example `fmp4-> fmp4 -> ts -> fmp4 -> fmp4`. We need to append an init segment on the first `fmp4`, the `ts`, and the `fmp4` right after the `ts`. If we don't set the `activeInitSegmentId_` to null here we will not append an init segment for the `fmp4` right after the `ts`